### PR TITLE
This Pull Request addresses the evolutionary change that adds the http_downstream scenario to the fluent-bit-test-suite for the Splunk plugin.

### PR DESCRIPTION
**Pull request by odonata**
This commit adds tests for the Splunk plugin in Fluent Bit. The test is under the directory called ‘http_downstream’ in scenarios.
    
    Three configuration files have been added in runtime for Fluent Bit:
    
            1.      splunk_on_http2_keepalive.yaml, with the configuration to test:
            •       http2: on
            •       net.keepalive: on
            •       tls: off
    
            2.      splunk_on_http2_keepalive_tlsON.yaml, with the configuration to test:
            •       http2: off
            •       net.keepalive: on
            •       tls: on
            •       tls.crt_file: ../certificate/certificate.pem
            •       tls.key_file: ../certificate/private_key.pem
    
            3.      splunk_on_http2_Nokeepalive.yaml, with the configuration to test:
            •       http2: on
            •       net.keepalive: off
            •       tls: off
            
            4.      splunk_on_http2ON_keepaliveON_tlsON.yaml, with the configuration to test:
            •       http2: on
            •       net.keepalive: on
            •       tls: off
    
    The four tests are created within logic_http_downstream_001.py using the FluentBitManager to dynamically create the connection at runtime and call the tests for each case, using 'pytest logic_http_downstream_001.py'

    The structure is as follows:
    
            •       scenario->http_downstream: is the parent directory of the test.
            •       scenario->http_downstream->certificate: is the directory where the certificates generated for TLS tests are located.
            •       scenario->http_downstream->config: is the directory that contains the YAML configuration files for Fluent Bit.
            •       scenario->http_downstream->test: is the directory that contains the Python program for tests with pytest.

### CODE REVIEW

File: scenarios/http_downstream/tests/logic_http_downstream_001.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import http.client, time, json, os, ssl, logging, requests
+
+from utils.fluent_bit_manager import FluentBitManager
+from utils.network import find_available_port
+from server.otlp_server import  data_storage
+
+logger = logging.getLogger(__name__)
+
+def create_connection_tlsON(server, port, cafile):
```
Comment (edsiper) [inline scenarios/http_downstream/tests/logic_http_downstream_001.py]: let's not use camel case, e.g: `def create_connection_tlsON()` -> `def create_connection_tls_on()`

File: scenarios/http_downstream/tests/logic_http_downstream_001.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        conn.request("POST", "/services/collector", body=json_payload, headers=headers)
+        response = conn.getresponse()
+        responses.append({
+            'status': response.status,
+            'reason': response.reason,
+            'data': response.read().decode()
+        })
+    return responses
+
+def test_splunk_on_http2_keepalive_tlsON():
```
Comment (edsiper) [inline scenarios/http_downstream/tests/logic_http_downstream_001.py]: please remove camel case

File: scenarios/http_downstream/tests/logic_http_downstream_001.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        for response in output:
+            assert response['status'] == 200
+            assert response['reason'] == 'OK'
+            assert response['data'] == '{"text":"Success","code":0}'
+    except Exception as e:
+        logger.error(f"An error occurred: {e}")
+        service.stop()
+        raise
+
+def test_splunk_on_http2_Nokeepalive():
```
Comment (edsiper) [inline scenarios/http_downstream/tests/logic_http_downstream_001.py]: please remove camel case

File: scenarios/http_downstream/tests/logic_http_downstream_001.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            assert response['reason'] == 'OK'
+            assert response['data'] == '{"text":"Success","code":0}'
+    except Exception as e:
+        logger.error(f"An error occurred: {e}")
+        service.stop()
+        raise
+
+
+
+def test_splunk_on_http2ON_keepaliveON_tlsON():
```
Comment (edsiper) [inline scenarios/http_downstream/tests/logic_http_downstream_001.py]: please remove camel case

File: scenarios/http_downstream/tests/logic_http_downstream_001.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        # find and set a listener port for the local server to receive the
+        # Fluent Bit output
+        self.test_suite_http_port = find_available_port(starting_port=50000)
+        os.environ['TEST_SUITE_HTTP_PORT'] = str(self.test_suite_http_port)
+        logger.info(f"test suite http port: {self.test_suite_http_port}")
+
+        # Start Fluent Bit
+        self.flb.start()
+
+    def run_splunk_TLS(self,server, port, cafile, num_requests):
```
Comment (edsiper) [inline scenarios/http_downstream/tests/logic_http_downstream_001.py]: please remove camel case

**Review by edsiper — CHANGES_REQUESTED** [review]
Thanks for the PR. Changes requested: 

- please remove all camel cases in names and file names, use `_` as a separator

**Comment by edsiper** [comment]
@odonata  Once is ready, please change it from draft to ready for review, thank you

File: scenarios/http_downstream/tests/logic_http_downstream_001.py (hunk 1 above)
Comment (odonata) [reply scenarios/http_downstream/tests/logic_http_downstream_001.py]: all requested changes have been applied.

**Comment by odonata** [comment]
all requested changes have been applied.

**Pull request merged**